medusa/providers/torrent/html/bjshare.py

```diff
@@ -0,0 +1,250 @@
+# coding=utf-8
+
+"""Provider code for BJ-Share."""
+
+from __future__ import unicode_literals
+
+import logging
+import re
+from collections import OrderedDict
+
+from medusa import tv
+from medusa.bs4_parser import BS4Parser
+from medusa.helper.common import convert_size, try_int
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+from requests.compat import urljoin
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class BJShareProvider(TorrentProvider):
+    """BJ-Share Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(BJShareProvider, self).__init__('BJ-Share')
+
+        # URLs
+        self.url = 'https://bj-share.me'
+        self.urls = {
+            'search': urljoin(self.url, "torrents.php")
+        }
+
+        # Credentials
+        self.enable_cookies = True
+        self.cookies = ''
+        self.required_cookies = ['session']
+
+        # Torrent Stats
+        self.minseed = None
+        self.minleech = None
+
+        # Miscellaneous Options
+        self.supports_absolute_numbering = True
+        self.max_back_pages = 2
+        self.units = ["B", "KB", "MB", "GB", "TB", "PB"]
+
+        # Proper Strings
+        self.proper_strings = ["PROPER", "REPACK", "REAL", "RERIP"]
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=30)
+
+        self.quality = OrderedDict([
```

Comment:
Why is this an OrderedDict and not just a dict?

Reply:
Otherwise Full HD could be parsed to Full 720p if the check for HD happens first...

Reply:
Remove this and just use replace.

Reply:
Ok, done.
Did this way because there could be more cases that i did not saw initially..